### ISSUE TITLE
Validation loss calculation (supposedly) breaks training 

### ISSUE THREAD

**Issue by 404ic** [neuroethology/MARS_Developer#31]
When running the training code, in a couple iterations it says the cost matrix is infeasible based on linear_sum_assignment in multibox_detection/loss.py When I remove all the code from train_detection.py that calculates the validation loss, the training works. 

Also noticed that when val_loss is computed, batch_size is passed as tf.get_shape(locs_v)[0] which doesn't give the batch size, but changing it to locs_v.get_shape().as_list()[0] should work.
